File: src/main/java/com/amberclient/modules/player/FastPlace.java
package com.amberclient.modules.player;

import com.amberclient.utils.module.Module;
import net.minecraft.client.MinecraftClient;
import net.minecraft.text.Text;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

public class FastPlace extends Module {
    public static final String MOD_ID = "amberclient-fastplace";
    public static final Logger LOGGER = LogManager.getLogger(MOD_ID);

    private static FastPlace instance;
    public static boolean isFastPlaceEnabled;

    public FastPlace() {
        super("FastPlace", "Blocks go brrrrrrr", "Player");
        instance = this;
    }

    public static FastPlace getInstance() {
        return instance;
    }

    @Override
    public void onEnable() {
        MinecraftClient client = MinecraftClient.getInstance();

        isFastPlaceEnabled = true;
        LOGGER.info(getName() + " module enabled");
    }

    @Override
    public void onDisable() {
        MinecraftClient client = MinecraftClient.getInstance();

        isFastPlaceEnabled = false;
        LOGGER.info(getName() + " module disabled");
    }
}

File: src/main/java/com/amberclient/modules/player/FastPlace.kt
package com.amberclient.modules.player

import com.amberclient.utils.module.Module
import net.minecraft.client.MinecraftClient
import net.minecraft.text.Text
import org.apache.logging.log4j.LogManager
import org.apache.logging.log4j.Logger
import kotlin.random.Random

class FastPlace : Module("FastPlace", "Blocks go brrrrrrr", "Player") {

    companion object {
        const val MOD_ID = "amberclient-fastplace"
        val LOGGER: Logger = LogManager.getLogger(MOD_ID)

        private var instance: FastPlace? = null

        @JvmField
        var isFastPlaceEnabled = false

        @JvmField
        var minDelay = 0

        @JvmField
        var maxDelay = 2

        @JvmField
        var stutterChance = 0.15f

        @JvmField
        var stutterMinDelay = 3

        @JvmField
        var stutterMaxDelay = 8

        @JvmStatic
        fun getRandomDelay(): Int {
            return if (Random.nextFloat() < stutterChance) {
                Random.nextInt(stutterMinDelay, stutterMaxDelay + 1)
            } else {
                Random.nextInt(minDelay, maxDelay + 1)
            }
        }

        fun getInstance(): FastPlace? = instance
    }

    init {
        instance = this
    }

    override fun onEnable() {
        isFastPlaceEnabled = true
    }

    override fun onDisable() {
        isFastPlaceEnabled = false
    }
}